modules/data.remote/R/soilgrids.R

```diff
@@ -0,0 +1,224 @@
+##' @title soilgrids.soc.extract
+##' @name  soilgrids.soc.extract
+##' 
+##' @param site_info A BETYdb site info dataframe containing at least each site ID, sitename, 
+##' latitude, longitude, and time_zone. e.g. c(site_qry$id, site_qry$sitename, site_qry$lon,
+##' site_qry$lat, site_qry$time_zone)
+##' @param outdir Directory to store SOC data.  Format *.Rdata
+##' @param verbose logical. If TRUE show the extraction progress bar
+##' 
+##' ##' @examples
+##' \dontrun{
+##' 
+##' # Example 1 - using the modex.bnl.gov BETYdb and site IDs to extract data
+##' db <- 'betydb'
+##' host_db <- 'modex.bnl.gov'
+##' db_port <- '5432'
+##' db_user <- 'bety'
+##' db_password <- 'bety'
+##' 
+##' bety <- list(user='bety', password='bety', host='modex.bnl.gov',
+##' dbname='betydb', driver=RPostgres::Postgres(),write=FALSE)
+##' 
+##' con <- DBI::dbConnect(drv=bety$driver, dbname=bety$dbname, host=bety$host, 
+##' password=bety$password, user=bety$user)
+##' 
+##' suppressWarnings(site_qry <- glue::glue_sql("SELECT *, ST_X(ST_CENTROID(geometry)) AS lon,
+##' ST_Y(ST_CENTROID(geometry)) AS lat FROM sites WHERE id IN ({ids*})",
+##' ids = c("676","622","678","766","764"), .con = con))
+##' 
+##' suppressWarnings(qry_results.1 <- DBI::dbSendQuery(con,site_qry))
+##' suppressWarnings(qry_results.2 <- DBI::dbFetch(qry_results.1))
+##' DBI::dbClearResult(qry_results.1)
+##' dbDisconnect(con)
+##' 
+##' site_info <- qry_results.2
+##' verbose <- TRUE
+##' system.time(result_soc <- soilgrids.soc.extract(site_info=site_info, verbose=verbose))
+##' result_soc
+##' 
+##' }
+##' 
+##' @importFrom reshape2 melt
+##' @importFrom terra vect project rast extract
+##' @importFrom stats qgamma optim
+##' @importFrom utils flush.console setTxtProgressBar txtProgressBar
+##' 
+##' @return list of two dataframes containing the mean SOC values per location and depth
```

Comment:
Is there an advantage to returning a list of two dataframes instead of one long dataframe (e.g. site_id, lat, lon, depth, mean, sd)?

Reply:
well, that's wide not long, but I guess it works